source/base/utilities.cc

```diff
@@ -484,6 +484,37 @@ namespace Utilities
 
 
 
+  template <typename Number>
+  Number
+  truncate_to_n_digits(const Number number, const unsigned int n_digits)
+  {
+    AssertThrow(n_digits >= 1, ExcMessage("invalid parameter."));
+
+    if (!(std::fabs(number) > std::numeric_limits<Number>::min()))
+      return number;
+
+    const int order =
+      static_cast<int>(std::floor(std::log10(std::fabs(number))));
+
+    const int max_order = static_cast<int>(
+      std::floor(std::log10(std::numeric_limits<Number>::max())));
+
+    const int shift = -order + static_cast<int>(n_digits) - 1;
+
+    AssertThrow(
+      shift <= max_order,
```

Comment:
You'll get a warning about an unused variable `max_order` in release mode because the variable is only used inside the assertion. Just inline the variable's value into this line.

Reply:
sorry, I don't know what inlining means in this context. Is AssertThrow correct?

Reply:
How about https://github.com/dealii/dealii/pull/8927/commits/b6f371ffde891122e9eb3baf9ef367ba43224272 ?

Reply:
```suggestion
    AssertThrow(
      shift <= static_cast<int>(
      std::floor(std::log10(std::numeric_limits<Number>::max()))),
```

Reply:
Instead of creating a named variable, just take the expression you use to define `max_order` and put it into the assertion here. Then there will simply be no variable called `max_order` at all.

Reply:
`AssertThrow` is also visible in Release mode. If you were using `Assert` instead we would need to be more careful.

Reply:
Oh, correct. But we should be using `Assert` here.